src/test/java/itest/UploadRecordingIT.java

```diff
@@ -92,26 +98,104 @@ public static void deleteRecording() throws Exception {
         webClient
                 .post(
                         String.format(
                                 "/api/v1/targets/%s/recordings/%s/upload",
-                                TARGET_ID, RECORDING_NAME))
+                                SELF_REFERENCE_TARGET_ID, RECORDING_NAME))
+                .send(
+                        ar -> {
+                            if (assertRequestStatus(ar, uploadRespFuture)) {
+                                MatcherAssert.assertThat(
+                                        ar.result().statusCode(), Matchers.equalTo(200));
+                            }
+                            uploadRespFuture.complete(ar.result().bodyAsString());
+                        });
+
+        final String expectedUploadResponse =
+                String.format("Uploaded: %s\nSet: %s", RECORDING_NAME, RECORDING_NAME);
+
+        MatcherAssert.assertThat(
+                uploadRespFuture.get().trim(), Matchers.equalTo(expectedUploadResponse));
+
+        // Confirm recording appears in Grafana
+        CompletableFuture<String> getRespFuture = new CompletableFuture<>();
+        webClient
+                .get(8080, "localhost", "/list")
                 .send(
                         ar -> {
-                            if (ar.failed()) {
-                                uploadRespFuture.completeExceptionally(ar.cause());
-                                return;
+                            if (assertRequestStatus(ar, getRespFuture)) {
+                                MatcherAssert.assertThat(
+                                        ar.result().statusCode(), Matchers.equalTo(200));
+                                MatcherAssert.assertThat(
+                                        ar.result().getHeader(HttpHeaders.CONTENT_TYPE.toString()),
+                                        Matchers.equalTo(HttpMimeType.JSON.mime()));
+                                getRespFuture.complete(ar.result().bodyAsString());
                             }
-                            HttpResponse<Buffer> result = ar.result();
-                            uploadRespFuture.complete(result.statusCode());
                         });
-        int statusCode = uploadRespFuture.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
-        MatcherAssert.assertThat(statusCode, Matchers.equalTo(200));
+
+        MatcherAssert.assertThat(
+                getRespFuture.get().trim(), Matchers.equalTo(String.format("%s", RECORDING_NAME)));
+
+        CompletableFuture<JsonArray> queryRespFuture = new CompletableFuture<>();
+
+        Calendar lastMinute = Calendar.getInstance();
+        lastMinute.add(Calendar.MINUTE, -1);
+        final DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS");
+        final String END = dateFormat.format(Calendar.getInstance().getTime());
+        final String START = dateFormat.format(lastMinute.getTime());
+
+        JsonObject query =
```

Comment:
I tried writing a query body with the docs I found below. How can I find the correct `panelId` and `targets` values?
This query gives me an empty array.

https://github.com/grafana/simple-json-datasource#query-api
https://oznetnerd.com/2018/04/17/writing-a-grafana-backend-using-the-simple-json-datasource-flask/#query-endpoint-deep-dive

Reply:
I'm not entirely sure how those queries work, to be honest, but I think you can find some inspiration in the jfr-datasource tests:

https://github.com/cryostatio/jfr-datasource/blob/main/src/test/java/io/cryostat/jfr/datasource/tests/DatasourceTest.java#L106

https://github.com/cryostatio/jfr-datasource/blob/main/src/test/resources/query.timeseries.input.txt

Reply:
I think I have the correct query format because I'm getting a non-empty response. 
```
{"target": "jdk.CPULoad.machineTotal", "datapoints": []}
```
The `datapoints` array is empty, though. Making a couple `GET` requests to the server and using `Thread.sleep(5000)` before uploading the recording to Grafana also doesn't add any datapoints. Any ideas?

Reply:
Interesting, the PR run seems to capture data:
```
Expected: <[{"datapoints":[],"target":"jdk.CPULoad.machineTotal"}]>
     but: was <[{"target":"jdk.CPULoad.machineTotal","datapoints":[[0.9300000071525574,1628278180851],[0.9845361113548279,1628278181823]]}]>
	at itest.UploadRecordingIT.shouldUploadRecordingToGrafana(UploadRecordingIT.java:204)
```
Looking into it.

Reply:
On this issue again...

A pretty surprising root cause behind is the time frame passed in the query is out of range. However, I have to set the time range to expand forward 1 month to get the data points. Quite odd... 

Still investigating...

Reply:
Actually, 24 hours is enough. This might be due to timezone difference.

Reply:
Nice! It was timezone issue. I adjusted it back to UTC-0 instead of local timezone.